### PR TITLE
add fence in write_tohost loop

### PR DESCRIPTION
**Pull request by timsnyder-siv**

### CODE REVIEW

File: p/riscv_test.h [hunk 1]
```diff
@@ -140,6 +140,7 @@ handle_exception:                                                       \
   1:    ori TESTNUM, TESTNUM, 1337;                                     \
   write_tohost:                                                         \
         sw TESTNUM, tohost, t5;                                         \
+        fence;                                                          \
```
Comment (pdonahue-ventana) [inline p/riscv_test.h:143]: Why is this necessary?

Comment (aswaterman) [reply p/riscv_test.h:143]: It’s not. After consulting with @timsnyder-siv, I am going to close this.

Comment (timsnyder-siv) [reply p/riscv_test.h:143]: @pdonahue-ventana , I talked with Andrew and it isn't.  Thanks for taking a look.
